meta/release_checklist.md

```diff
@@ -0,0 +1,42 @@
+# Release Checklist
```

Comment:
Should we put this also in `docs` for now? I hesitant to introduce a new directory.

Reply:
I viewed `docs/` as the pieces of the readme, not a general place to put documentation. But I don't mind moving it there :-)

Reply:
Let's move it there for now until we figured out a better place. I'm mainly worried about the naming `meta` :-)